### PR TITLE
merge host info from kvstore

### PR DESCRIPTION
**Pull request by andychow01**
fix bug:
    https://github.com/vesoft-inc/nebula/issues/532

### CODE REVIEW

File: src/meta/ActiveHostsMan.cpp [hunk 1]
```diff
@@ -120,6 +120,26 @@ void ActiveHostsMan::cleanExpiredHosts() {
...
+    // merge host info from kvstore
+    if (kvstore_ != nullptr) {
+        const auto& prefix = MetaServiceUtils::hostPrefix();
+        std::unique_ptr<kvstore::KVIterator> iter;
+        auto ret = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+        if (ret == kvstore::ResultCode::SUCCEEDED) {
+            while (iter->valid()) {
+                auto host = MetaServiceUtils::parseHostKey(iter->key());
+                if (iter->val() == MetaServiceUtils::hostValOnline()) {
+                    bool found = hostsMap_.find({host.ip, host.port}) == hostsMap_.end();
```
Comment (critical27) [inline src/meta/ActiveHostsMan.cpp]: Maybe you need a lock here

Comment (critical27) [reply src/meta/ActiveHostsMan.cpp]: Also maybe just use if (hostsMap_.find({host.ip, host.port}) == hostsMap_.end()) here, you name it "found", which means "not found in memory", it confuses.

Comment (critical27) [reply src/meta/ActiveHostsMan.cpp]: Here is another case, what if it is offline in kvstore, but online in memory, should we handle it? If we don't, those host will stay offline forever, the result of getActiveHosts and "show hosts" will be different.

File: src/meta/test/ActiveHostsManTest.cpp [hunk 2]
```diff
@@ -37,6 +37,56 @@ TEST(ActiveHostsManTest, NormalTest) {
     }
 }
 
+TEST(ActiveHostsManTest, NormalTest2) {
```
Comment (spacewalkman) [inline src/meta/test/ActiveHostsManTest.cpp]: Should give it more meaningful test name like `MergeHostTest` ? WDYT?

File: src/meta/ActiveHostsMan.cpp [hunk 3]
```diff
@@ -120,6 +120,27 @@ void ActiveHostsMan::cleanExpiredHosts() {
...
+        auto ret = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+        if (ret == kvstore::ResultCode::SUCCEEDED) {
+            while (iter->valid()) {
+                auto host = MetaServiceUtils::parseHostKey(iter->key());
+                if (iter->val() == MetaServiceUtils::hostValOnline()) {
+                    folly::RWSpinLock::ReadHolder rh(&lock_);
+                    bool notFound = hostsMap_.find({host.ip, host.port}) == hostsMap_.end();
+                    if (notFound) {
+                        data.emplace_back(MetaServiceUtils::hostKey(host.ip, host.port),
+                                MetaServiceUtils::hostValOffline());
```
Comment (dangleptr) [inline src/meta/ActiveHostsMan.cpp]: Alignment

File: src/meta/test/ActiveHostsManTest.cpp [hunk 4]
```diff
@@ -37,6 +37,56 @@ TEST(ActiveHostsManTest, NormalTest) {
...
+    fs::TempDir rootPath("/tmp/ActiveHostsMergeTest.XXXXXX");
+    std::unique_ptr<kvstore::KVStore> kv(TestUtils::initKV(rootPath.path()));
+    {
+        std::vector<kvstore::KV> data;
+        for (auto i = 0; i < 3; i++) {
+            data.emplace_back(MetaServiceUtils::hostKey(0, i),
+                MetaServiceUtils::hostValOnline());
```
Comment (dangleptr) [inline src/meta/test/ActiveHostsManTest.cpp]: Alignment

File: src/meta/test/ActiveHostsManTest.cpp (hunk 2 above)
Comment (dangleptr) [reply src/meta/test/ActiveHostsManTest.cpp]: +1

File: src/meta/test/ActiveHostsManTest.cpp [hunk 5]
```diff
@@ -37,6 +37,56 @@ TEST(ActiveHostsManTest, NormalTest) {
...
+        } else {
+            ++offlineNum;
+        }
+        iter->next();
+    }
+    ASSERT_EQ(3, onlineNum);
+    ASSERT_EQ(0, offlineNum);
+
+    ActiveHostsMan ahMan(1, 1, kv.get());
+    sleep(3);
```
Comment (dangleptr) [inline src/meta/test/ActiveHostsManTest.cpp]: sleep(intervalSeconds + 1)

File: src/meta/test/ActiveHostsManTest.cpp (hunk 4 above)
Comment (andychow01) [reply src/meta/test/ActiveHostsManTest.cpp]: fixed

File: src/meta/ActiveHostsMan.cpp (hunk 3 above)
Comment (andychow01) [reply src/meta/ActiveHostsMan.cpp]: fixed

File: src/meta/test/ActiveHostsManTest.cpp (hunk 2 above)
Comment (andychow01) [reply src/meta/test/ActiveHostsManTest.cpp]: fixed

File: src/meta/test/ActiveHostsManTest.cpp (hunk 5 above)
Comment (andychow01) [reply src/meta/test/ActiveHostsManTest.cpp]: fixed

**Pull request merged**